RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -269,11 +271,11 @@ double MahiFit::minimize() const {
 }
 
 void MahiFit::updatePulseShape(double itQ, FullSampleVector &pulseShape, FullSampleVector &pulseDeriv,
-			       FullSampleMatrix &pulseCov) const {
+			       FullSampleMatrix &pulseCov,
+			       const HcalTimeSlew* hcalTimeSlew_delay) const {
   
   float t0=meanTime_;
-  if (applyTimeSlew_) 
-    t0+=HcalTimeSlew::delay(std::max(1.0, itQ), slewFlavor_);
+  if (applyTimeSlew_) t0+=hcalTimeSlew_delay->delay(std::max(1.0, itQ), slewFlavor_);
```

Comment:
Can we please retain the previous behavior? Possibly by adding a method like setPulseShapeTemplate [here](https://github.com/cms-sw/cmssw/blob/master/RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc#L65)?

Reply:
As long as this works I can mess with the technical parts later.